File: typescript-sdk/src/sdk/models/operations/postroomsroomid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostRoomsRoomIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=roomId" })
  roomId: string;
}


export class PostRoomsRoomIdRequests extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/xml" })
  applicationXML: Uint8Array;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  updateRoom?: shared.UpdateRoom;

  @SpeakeasyMetadata({ data: "request, media_type=multipart/form-data" })
  object?: Record<string, any>;
}


export class PostRoomsRoomIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostRoomsRoomIdPathParams;

  @SpeakeasyMetadata()
  request?: PostRoomsRoomIdRequests;
}


export class PostRoomsRoomIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  room?: shared.Room;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/postroomsroomid.go
package operations

import (
	"openapi/pkg/models/shared"
)

type PostRoomsRoomIDPathParams struct {
	RoomID string `pathParam:"style=simple,explode=false,name=roomId"`
}

type PostRoomsRoomIDRequests struct {
	ApplicationXML []byte                 `request:"mediaType=application/xml"`
	UpdateRoom     *shared.UpdateRoom     `request:"mediaType=application/json"`
	Object         map[string]interface{} `request:"mediaType=multipart/form-data"`
}

type PostRoomsRoomIDRequest struct {
	PathParams PostRoomsRoomIDPathParams
	Request    *PostRoomsRoomIDRequests
}

type PostRoomsRoomIDResponse struct {
	ContentType string
	Error       *shared.Error
	Room        *shared.Room
	StatusCode  int64
}
